semgrep-core/src/matching/Generic_vs_generic.ml

```diff
@@ -485,7 +493,25 @@ let rec m_name a b =
       (* this will handle metavariables in Id *)
       m_ident_and_id_info (a1, a2) (b1, b2)
   | G.Id ((str, tok), _info), G.IdQualified _ when MV.is_metavar_name str ->
-      envf (str, tok) (MV.N b)
+      (* If `is_resolved` is true, then we got the target `IdQualified` from a
```

Comment:
This explanation still left me confused 😅. It's definitely tricky to explain. I think it would be helpful if you zoomed out in the beginning. It's not intuitively clear to me that it's bad for an identifier to have a nonsensical range, so I could have used some context there.